pkg/controllers/management/kontainerdrivermetadata/data_getter.go

```diff
@@ -105,47 +105,31 @@ func GetK8sVersionInfo(rancherVersion string, rkeSysImages map[string]v3.RKESyst
 		}
 	}
 	for majorVersion, k8sVersion := range maxVersionForMajorK8sVersion {
-		k8sVersionRKESystemImages[k8sVersion] = rkeSysImages[k8sVersion]
-		k8sVersionSvcOptions[k8sVersion] = svcOptions[majorVersion]
-	}
-	return k8sVersionRKESystemImages, k8sVersionSvcOptions
-}
-
-func GetRKEK8sServiceOptionsWindows(k8sVersion string, svcOptionLister v3.RKEK8sServiceOptionLister, svcOptions v3.RKEK8sServiceOptionInterface) (*v3.KubernetesServicesOptions, error) {
-	names := []string{
-		getWindowsName(k8sVersion),
-		getWindowsName(util.GetTagMajorVersion(k8sVersion)),
-	}
-	var k8sSvcOption *v3.KubernetesServicesOptions
-	for _, name := range names {
-		obj, err := svcOptionLister.Get(namespace.GlobalNamespace, name)
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				return k8sSvcOption, err
-			}
+		// detect the svc options existing or not
+		if _, exist := svcOptions[majorVersion]; !exist {
```

Comment:
we should first search for `svcOptions[k8sVersion]`, if it exists, store it for that version. If not, search for `majorVersion`. This is because we're supporting service options for both

Reply:
Updated